zerver/tests/test_messages.py

```diff
@@ -587,6 +591,82 @@ def test_applying_markdown(self):
         self.assertEqual(message.rendered_content, expected_content)
         self.assertEqual(message.rendered_content_version, bugdown.version)
 
+
+    def test_reaction(self):
```

Comment:
In addition to these unit tests, I'd add an end-to-end test that actually verifies the actual old messages API endpoint returns the reactions... That way, if we refactor, we have a test we can count on to ensure everything is correct.